MeshGeoToolsLib/CreateSearchLength.h

```diff
@@ -11,22 +11,25 @@
 
 #pragma once
 
+#include <memory>
+
 #include "BaseLib/ConfigTree.h"
 
 #include "MeshGeoToolsLib/HeuristicSearchLength.h"
 #include "MeshGeoToolsLib/SearchLength.h"
 
 namespace MeshGeoToolsLib
 {
-MeshGeoToolsLib::SearchLength createSearchLengthAlgorithm(
+std::unique_ptr<MeshGeoToolsLib::SearchLength> createSearchLengthAlgorithm(
```

Comment:
OK, I see. Looks like another of these C++ quirks: At least to me it's not obvious why C++ supports such constructions: virtual methods which are dispatched dynamically only if pointers or references are used.

Maybe then `HeuristicSearchLength` shouldn't be derived from `SearchLength`, but both should have a common abstratct base class.